contracts/modules/ApprovedTransfer.sol

```diff
@@ -100,6 +115,31 @@ contract ApprovedTransfer is BaseTransfer {
         onlyAuthorisedContractCall(_wallet, _contract)
     {
         doApproveTokenAndCallContract(_wallet, _token, _spender, _amount, _contract, _data);
+        LimitUtils.resetDailySpent(limitStorage, _wallet);
+    }
+
+    /**
+     * @dev Changes the daily limit. The change is immediate.
+     * @param _wallet The target wallet.
+     * @param _newLimit The new limit.
+     */
+    function changeLimit(address _wallet, uint256 _newLimit) external onlyWalletModule(_wallet) onlyWhenUnlocked(_wallet) {
+        uint128 targetLimit = LimitUtils.safe128(_newLimit);
+        // solium-disable-next-line security/no-block-members
+        ILimitStorage.Limit memory newLimit = ILimitStorage.Limit(targetLimit, targetLimit, LimitUtils.safe64(now));
+        // solium-disable-next-line security/no-block-members
+        ILimitStorage.DailySpent memory resetDailySpent = ILimitStorage.DailySpent(uint128(0), LimitUtils.safe64(now));
```

Comment:
I think `DailySpent.periodEnd` could be set to 0 (both here and in `LimitUtils.resetDailySpent()`) since, unlike `Limit.changeAfter`, it is not used in events. That would make the treatment of this param slightly different to the way `Limit.changeAfter` is set but considering the number of times `resetDailySpent()` can be called, I think the gas savings would be worth this small inconsistency IMO.

Reply:
Changed.